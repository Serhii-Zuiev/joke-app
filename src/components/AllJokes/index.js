export { default as AllJokesContainer } from "./AllJokesContainer";
export { default as JokesList } from "./JokesList";
export { default as JokeItem } from "./JokeItem";