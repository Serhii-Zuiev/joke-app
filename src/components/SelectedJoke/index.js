export { default as SelectedJoke } from "./SelectedJoke";