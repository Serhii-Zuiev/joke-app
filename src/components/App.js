import React from "react";
import styled from "styled-components";
import AllJokes from "./AllJokes/AllJokesContainer";
import SelectedJoke from "./SelectedJoke/SelectedJoke";

function App() {
  return (
    <StyledAppContainer>
      <StyledHeader>
        <StyledAppHeading>
          <span role="img" aria-label="face emoji">
            ðŸ˜„
          </span>
          Joke app
        </StyledAppHeading>
      </StyledHeader>

      <AllJokes />
      {/* <SelectedJoke /> */}
    </StyledAppContainer>
  );
}

export default App;

const StyledAppContainer = styled.div`
  background-color: #eee;
`;

const StyledHeader = styled.header`
  height: 70px;
  width: 90%;
  margin: 0 auto;
`;

const StyledAppHeading = styled.h1`
  font-size: 26px;
  color: #333;
  padding: 16px 0;
  letter-spacing: 1px;
  text-align: center;
  border-bottom: 2px solid #777;
`;
