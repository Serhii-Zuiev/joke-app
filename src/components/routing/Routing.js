import React from "react";
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from "react-router-dom";
import { AllJokesContainer } from '../AllJokes'
import { SelectedJoke } from '../SelectedJoke'

const Routing = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={AllJokesContainer} />
        <Route path="/joke/:id" component={SelectedJoke} />
        <Redirect to="/" />
      </Switch>
    </Router>
  );
};

export default Routing;
