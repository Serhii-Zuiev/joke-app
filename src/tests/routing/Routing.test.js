import React from "react";
import { Routing } from "../../components/routing";

describe("<Routing/> unit test", () => {
    it("should render Routing component", () => {
        const component = shallow(<Routing />);
        expect(component).toMatchSnapshot();
    });
});
