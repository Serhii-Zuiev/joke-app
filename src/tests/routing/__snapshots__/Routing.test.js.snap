// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routing/> unit test should render Routing component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/joke/:id"
    />
    <Redirect
      to="/"
    />
  </Switch>
</BrowserRouter>
`;
